input/config.js

```diff
@@ -1,103 +1,161 @@
+ * BEGIN CONFIG
+ *************************************************************/
+
+// image width in pixels
+const width = 1000;
+// image height in pixels
+const height = 1000;
+// description for NFT in metadata file
+const description = "This is an NFT made by the coolest generative code.";
+// base url to use in metadata file
+// the id of the nft will be added to this url, in the example e.g. https://hashlips/nft/1 for NFT with id 1
+const baseImageUri = "https://hashlips/nft";
+// id for edition to start from
+const startEditionFrom = 1;
+// amount of NFTs to generate in edition
+const editionSize = 10;
+// prefix to add to edition dna ids (to distinguish dna counts from different generation processes for the same collection)
+const editionDnaPrefix = 0
+
+// create required weights
+// for each weight, call 'addRarity' with the id and from which to which element this rarity should be applied
+let rarityWeights = [
+  addRarity('super_rare', 1, 1),
+  addRarity('rare', 2, 5),
+  addRarity('original', 5, 10)
+];
+
+// create required layers
+// for each layer, call 'addLayer' with the id and optionally the positioning and size
+// the id would be the name of the folder in your input directory, e.g. 'ball' for ./input/ball
 const layers = [
-  {
-    elements: {
-      original: getElements(`${dir}/ball/original`),
-      rare: getElements(`${dir}/ball/rare`),
-      super_rare: getElements(`${dir}/ball/super_rare`),
-    },
-    position: { x: 0, y: 0 },
-    size: { width: width, height: height },
-  },
-  {
-    elements: {
-      original: getElements(`${dir}/eye color/original`),
-      rare: getElements(`${dir}/eye color/rare`),
-      super_rare: getElements(`${dir}/eye color/super_rare`),
-    },
-    position: { x: 0, y: 0 },
-    size: { width: width, height: height },
-  },
-  {
-    elements: {
-      original: getElements(`${dir}/iris/original`),
-      rare: getElements(`${dir}/iris/rare`),
-      super_rare: getElements(`${dir}/iris/super_rare`),
-    },
-    position: { x: 0, y: 0 },
-    size: { width: width, height: height },
-  },
-  {
-    elements: {
-      original: getElements(`${dir}/shine/original`),
-      rare: getElements(`${dir}/shine/rare`),
-      super_rare: getElements(`${dir}/shine/super_rare`),
-    },
-    position: { x: 0, y: 0 },
-    size: { width: width, height: height },
-  },
-  {
-    elements: {
-      original: getElements(`${dir}/bottom lid/original`),
-      rare: getElements(`${dir}/bottom lid/rare`),
-      super_rare: getElements(`${dir}/bottom lid/super_rare`),
-    },
-    position: { x: 0, y: 0 },
-    size: { width: width, height: height },
-  },
-  {
-    elements: {
-      original: getElements(`${dir}/top lid/original`),
-      rare: getElements(`${dir}/top lid/rare`),
-      super_rare: getElements(`${dir}/top lid/super_rare`),
-    },
-    position: { x: 0, y: 0 },
-    size: { width: width, height: height },
-  },
+  addLayer('ball'),
```

Comment:
`addLayer` is now much more readable. It takes the provided layer name in combination with the previously defined rarities to find the necessary elements. It also uses position and size optionally - in most cases users can just use same size layers (like in the examples).

Reply:
Please make sure this works fine and add it to the readme file then we can pull it in.

Reply:
@HashLips did you identify any issues here? Should be working fine.
i.e. adding `addLayer('iris', { x: 200, y: 200 }, { width: 250, height: 250 }),` leads to
![image](https://user-images.githubusercontent.com/2007082/131816911-ed3c377a-6266-4dc0-8e34-3d2521b3e83d.png)
as expected.
Adding one line with parameters as example and another phrase in readme.